app/controllers/benefits/LivingAccommodationBenefitAmountController.scala

```diff
@@ -0,0 +1,112 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.benefits
+
+import config.{AppConfig, ErrorHandler}
+import controllers.benefits.routes.CompanyCarFuelBenefitsController
+import controllers.employment.routes.CheckYourBenefitsController
+import controllers.predicates.{AuthorisedAction, InYearAction}
+import forms.{AmountForm, FormUtils}
+import models.mongo.EmploymentCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.{EmploymentSessionService, RedirectService}
+import services.RedirectService.{ConditionalRedirect, EmploymentBenefitsType, redirectBasedOnCurrentAnswers}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.benefits.LivingAccommodationBenefitAmountView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class LivingAccommodationBenefitAmountController @Inject()(implicit val cc: MessagesControllerComponents,
+                                                           authAction: AuthorisedAction,
+                                                           inYearAction: InYearAction,
+                                                           appConfig: AppConfig,
+                                                           livingAccommodationBenefitAmountView: LivingAccommodationBenefitAmountView,
+                                                           val employmentSessionService: EmploymentSessionService,
+                                                           errorHandler: ErrorHandler,
+                                                           ec: ExecutionContext,
+                                                           clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper with FormUtils{
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getAndHandle(taxYear, employmentId) { (optCya, prior) =>
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+          val cyaAmount = cya.employment.employmentBenefits.flatMap(_.accommodationRelocationModel.flatMap(_.accommodation))
+
+          val form = fillFormFromPriorAndCYA(buildForm(user.isAgent), prior, cyaAmount, employmentId)(
+            employment =>
+              employment.employmentBenefits.flatMap(_.benefits.flatMap(_.accommodation))
```

Comment:
should this be getting the accommodation in the new accommodation relocation model instead of the one in the regular benefits model?

Reply:
oh yeah it is, sorry